pkg/clients/resourcerecordset/resourcerecordset.go

```diff
@@ -0,0 +1,177 @@
+package resourcerecordset
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/crossplane/provider-aws/apis/network/v1alpha3"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	//RRSetNotFound is the error code that is returned if RRSet is present
+	RRSetNotFound = "InvalidRRSetName.NotFound"
+)
+
+// Client defines ResourceRecordSet operations
+type Client interface {
+	ChangeResourceRecordSetsRequest(input *route53.ChangeResourceRecordSetsInput) route53.ChangeResourceRecordSetsRequest
+	ListResourceRecordSetsRequest(input *route53.ListResourceRecordSetsInput) route53.ListResourceRecordSetsRequest
+}
+
+// NotFound will be raised when there is no ResourceRecordSet
+type NotFound struct{}
+
+func (err *NotFound) Error() string {
+	return fmt.Sprint(RRSetNotFound)
+}
+
+// NewClient creates new AWS client with provided AWS Configuration/Credentials
+func NewClient(config *aws.Config) Client {
+	return route53.New(*config)
+}
+
+// GenerateChangeResourceRecordSetsInput prepares input for a ChangeResourceRecordSetsInput
+func GenerateChangeResourceRecordSetsInput(p *v1alpha3.ResourceRecordSetParameters, action route53.ChangeAction) *route53.ChangeResourceRecordSetsInput {
+	var ttl *int64
+	if p.TTL == nil {
+		num := int64(300)
+		ttl = &num
+	} else {
+		ttl = p.TTL
+	}
+
+	resourceRecords := make([]route53.ResourceRecord, 0)
+	for _, r := range p.Records {
+		record := r
+		resourceRecords = append(resourceRecords, route53.ResourceRecord{
+			Value: &record,
+		})
+	}
+
+	resourceRecordSet := route53.ResourceRecordSet{
+		Name:            p.Name,
+		Type:            route53.RRType(aws.StringValue(p.Type)),
+		TTL:             ttl,
+		ResourceRecords: resourceRecords,
+	}
+	changeBatch := &route53.ChangeBatch{
+		Changes: []route53.Change{
+			{
+				Action:            action,
+				ResourceRecordSet: &resourceRecordSet,
+			},
+		},
+	}
```

Comment:
Nit: Is there much value in creating `changeBatch` and `resourceRecordSet` as distinct variables vs just declaring them inline below?

Reply:
As far as I know, not in particular. I decided to keep it that way just for the sake of increased readability. But have made the changed that to inline declaration now.